### ISSUE TITLE
bytes, strings: Compare tests do not modify input alignments

### ISSUE THREAD

**Issue by mundaym** [golang/go#26129]
I can't see any tests varying the alignment of `Compare` arguments. Since `Compare` is written in assembly on most platforms (in `internal/bytealg`) we should probably be checking that it works even if the inputs are unaligned.

**Comment by mundaym** [comment]
@LotusFenn Yes, bytes/compare_test.go and strings/compare_test.go. An additional short test in each that just varies the input alignment from, say, 0 to 16 (inclusive) would be great. Thanks!

**Comment by agnivade** [comment]
@mundaym - The Compare function in `strings` does not use assembly and has a simple implementation.

```
	if a == b {
		return 0
	}
	if a < b {
		return -1
	}
	return +1
```

It even has a note from @rsc saying - 

```
	// NOTE(rsc): This function does NOT call the runtime cmpstring function,
	// because we do not want to provide any performance justification for
	// using strings.Compare. Basically no one should use strings.Compare.
```

Would you still want to test unaligned input for strings/compare_test.go ?

**Comment by mundaym** [comment]
@agnivade Good point. I still think it would be worthwhile though. The compiler could do the substitution with `runtime.cmpstring` or other alignment-sensitive vector code under the hood (as rsc alluded to in that comment).
